react-components/src/components/RevealToolbar/SettingsContainer/HighFidelityContainer.tsx

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { Menu } from '@cognite/cogs.js';
+import { type ReactElement, useEffect, useMemo } from 'react';
+import { useReveal } from '../../RevealContainer/RevealContext';
+import { type ResolutionOptions } from '@cognite/reveal';
+import { type HighFidelityProps } from './types';
+
+export const HighFidelityContainer = ({
+  isHighFidelityMode,
+  setHighFidelityMode
```

Comment:
Why can't this be a state within this component?

Reply:
The state will get reset when `SettingContainer` is re-mounted when other Button in toolbar is selected or when user clicks outside `SettingContainer` which will unmount the component.